Code for loading in the raw data and labels.

rezhv opened the pull request.

rezhv committed:
Code for loading in the raw data and labels.

AlynaGray commented:
Looks good! I'll add some of my thoughts in the other comments.

AlynaGray commented:
```python
X_train_b = np.concatenate((X_e1_p11, X_e1_p12, X_e2_p11, X_e2_p12))
X_test_b = np.concatenate((X_e2_p21, X_e2_p22))
```
Instead of splitting the dataset into a test and train set here. I think that we should split it later. We might want to use a specific algorithm to split the set in order to ensure a certain distribution of labels, split at different percentages, or do something like k-fold validation.

AlynaGray commented:
```python
X_train_c1 = np.concatenate(())
```
This part is empty. Are you intentionally leaving out data? I can see that you load `X_train_c1` and `X_train_c2` later
```python
X_train_c1 = np.load('drive/MyDrive/Colab Notebooks/bst227 project/X_train_c1.npy')
X_train_c2 = np.load('drive/MyDrive/Colab Notebooks/bst227 project/X_train_c2.npy')
```
Since the purpose of this notebook is to parse the dateset into npy files, I don't think we can rely on these files already existing. I think that these lines should be replaced with loading the data from the folders.

AlynaGray commented:
```python
X_train_c = np.concatenate((X_e1_p11_c, X_e1_p12_c, X_e1_p21_c, X_e1_p22_c))
```
```python
del X_e1_p11_c,X_e1_p12_c,X_e1_p21_c,X_e1_p22_c
```
```python
X_train_c = np.concatenate((X_train_c, X_e2_p11_c, X_e2_p12_c, X_e2_p21_c, X_e2_p22_c))
```
```python
del X_e2_p11_c,X_e2_p12_c,X_e2_p21_c,X_e2_p22_c
```
Related to my previous comment, experiment 3 data is not added to `X_train_c`. In addition, I think that you are create `X_train_c` piecewise to save memory, and I think that it would be more clear to put all the code from creating `X_train_c` from the components into the same box. Finally, `X_train_c` is never saved. I would include commented out code for saving it like with allele B.

AlynaGray commented:
```python
X_train_c_a3 = np.concatenate((X_train_c1[:, 2, :, :, :], X_train_c2[:, 2, :, :, :]))
np.save('drive/MyDrive/Colab Notebooks/bst227 project/X_train_c_a3.npy', X_train_c_a3)
del X_train_c_a3
```
```python
X_train_c_a4 = np.concatenate((X_train_c1[:, 3, :, :, :], X_train_c2[:, 3, :, :, :]))
np.save('drive/MyDrive/Colab Notebooks/bst227 project/X_train_c_a4.npy', X_train_c_a4)
del X_train_c_a4
```
I'm not sure what `X_train_c1` and `X_train_c2` are, but I think that this code can be removed since we should be saving `X_train_c`. This code also only saves half of the images in those two tensors and I don't see why you are doing that.

AlynaGray commented:
```python
X_e1_p11_c = create_dataset(df, img_e1_p11_c, 1, '1.1', assign=False)
X_e1_p12_c = create_dataset(df, img_e1_p12_c, 1, '1.2', assign=False)
X_e1_p21_c = create_dataset(df, img_e1_p21_c, 1, '2.1', assign=False)
X_e1_p22_c = create_dataset(df, img_e1_p22_c, 1, '2.2', assign=False)

X_e2_p11_c = create_dataset(df, img_e2_p11_c, 2, '1.1', assign=False)
X_e2_p12_c = create_dataset(df, img_e2_p12_c, 2, '1.2', assign=False)
X_e2_p21_c = create_dataset(df, img_e2_p21_c, 2, '2.1', assign=False)
X_e2_p22_c = create_dataset(df, img_e2_p22_c, 2, '2.2', assign=False)

X_e3_p11_c = create_dataset(df, img_e3_p11_c, 3, '1.1', assign=False)
X_e3_p12_c = create_dataset(df, img_e3_p12_c, 3, '1.2', assign=False)
X_e3_p21_c = create_dataset(df, img_e3_p21_c, 3, '2.1', assign=False)
X_e3_p22_c = create_dataset(df, img_e3_p22_c, 3, '2.2', assign=False)
```
The y labels for allele C aren't being saved which means that they aren't being concated together and saved later.

AlynaGray commented:
As for the Tensorflow code, I think that we should archive them for future reference, but the code for parsing the dataset should be in separate file. Perhaps we can rename this file Fish_Data_Training_Tensorflow.ipynb, put it in the Classifier/exploratory-research, and create a new file parse_dataset.ipynb for our parsing code.

AlynaGray commented:
I just realized that we already had a copy of this code in Classifier/exploratory-research. The only differences in this code is that this section is commented out
```python
# np.save('y_train_b.npy', y_train_b)
# np.save('y_test_b.npy', y_test_b)
````
```python
# np.save('X_train_b.npy', y_train_b)
# np.save('X_test_b.npy', y_test_b)
```
and the tensors are being created and saved here
```python
x_train_b = np.concatenate([X_train_b,X_test_b])
y_train_b = np.concatenate([y_train_b,y_test_b])
```
```python
np.save("x_train_b.npy",x_train_b)
np.save("y_train_b.npy",y_train_b)
```
```python
y_train_b.shape
```
I support deleting the file in Classifier/exploratory-research and replacing it with this one.

AlynaGray commented:
I don't see the place in the code where we strip out the fishes with an unknown label. I suppose it's not in this version?

rezhv committed:
Updated

AlynaGray commented:
The updated branch looks good to me! I just noticed that the plot examples part needs to change the variable form `X_train_b` to `X_train`.
```python
import matplotlib.pyplot as plt

plot_fish(X_train_b, 0)
plot_fish(X_train_b, 1)
plot_fish(X_train_b, 2)
plot_fish(X_train_b, 3)
```

As for the code for transfer learning. I think that we can delete all that because a copy of the code is already in Classifier/exploratory-research/Fish_Data_Training.ipynb. Afterwards, we should rename this file to something like parse_dataset.ipynb

Commit:
Clean up data extraction

I made three changes:
1. Remove training code
2. Remove unlabeled fished from allele b
3. Fix code that plots examples of fishes

Commit:
Rename Fish_Data_Training.ipynb to Data_Processing.ipynb

This name better reflects the purpose of the file since we are using it to
extract data, not train a model.

Commit:
Normalize fish images to 0-1
